### PR TITLE
[IOTDB-5881] Support path contains filter in show timeseries and show devices

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

This PR refactors the data structure of the metadata query filter criteria, using a SchemaFilter abstraction class to encapsulate it for better extensibility.

* Path contains filter is supported in `show timeseries`, `show devices ` and `count timeseries`.
* Modify SQL for tag filter. Use TAGS(key) to identify the tag need to be filtered.

## Example

Dataset:
image

* show timeseries where timeseries contains 
image

* show devices where device contains 
image

* show timeseries where TAGS()
image

* count timeseries where timeseries contains 
image

### CODE REVIEW

**Review by SzyWilliam — COMMENTED** [review]
Bravo work!

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/impl/PathContainsFilter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.apache.iotdb.db.mpp.plan.schemafilter.SchemaFilterType;
+import org.apache.iotdb.db.mpp.plan.schemafilter.SchemaFilterVisitor;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public class PathContainsFilter extends SchemaFilter {
+  private final String pathContains;
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/impl/PathContainsFilter.java:31]: Rename to ```containedString```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/SchemaFilter.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.apache.iotdb.db.mpp.plan.schemafilter.impl.PathContainsFilter;
+import org.apache.iotdb.db.mpp.plan.schemafilter.impl.TagFilter;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public abstract class SchemaFilter {
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/SchemaFilter.java:29]: Move to metadata package.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/impl/PathContainsFilter.java (hunk 1 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/impl/PathContainsFilter.java]: done

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/SchemaFilter.java (hunk 2 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/schemafilter/SchemaFilter.java]: done

**Review by MarcosZyk — APPROVED** [review]

**Pull request merged**